js/service.js

```diff
@@ -317,7 +318,16 @@ let Service = {
     },
     put: function(updatedProfile) {
       if (updatedProfile.newsletter) {
-        Service.signUp(user.email, window.location.toString());
+        basketSignup(
+          {
+            email: user.email,
+            privacy: true
+          },
+          () => {},
```

Comment:
it doesn't feel like we're actually doing anything meaningful with onSuccessCallback and onFailCallback here: do we need them? If not, it might be better to outright remove them from the basket function.